File: app/src/main/java/com/androidstudy/mpesa/utils/AppUtils.java
package com.androidstudy.mpesa.utils;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

import java.util.Calendar;
import java.util.Date;
import java.util.UUID;

public class AppUtils {


    public static String UUID() {
        return UUID.randomUUID().toString();
    }

    public static String getPassKey() {
        return "bfb279f9aa9bdbcf158e97dd71a467cd2e0c893059b10f78e6b72ada1ed2c919";
    }

    public static void saveAccessToken(Context context, String accessToken){
        SharedPreferences mSettings = PreferenceManager.getDefaultSharedPreferences(context);
        SharedPreferences.Editor editor = mSettings.edit();

        Calendar cal = Calendar.getInstance();
        cal.add(Calendar.HOUR, 1);
        Date oneHourAfter = cal.getTime();

        editor.putString("accessToken", accessToken);
        editor.putString("expiry_date", oneHourAfter.toString());

        editor.apply();
    }

    public static String getAccessToken(Context context){
        SharedPreferences mSettings = PreferenceManager.getDefaultSharedPreferences(context);

        return mSettings.getString("accessToken", "");
    }

    //TODO(check for if access token expired)
    public static boolean expired(Context context){
        SharedPreferences mSettings = PreferenceManager.getDefaultSharedPreferences(context);

        return true;
    }




}

File: app/src/main/java/com/androidstudy/mpesa/utils/AppUtils.kt
package com.androidstudy.mpesa.utils

import android.content.Context
import android.content.Context.*
import com.androidstudy.mpesa.BuildConfig
import java.util.*

object AppUtils {
    fun UUID(): String {
        return UUID.randomUUID().toString()
    }

    val passKey: String
        get() = "bfb279f9aa9bdbcf158e97dd71a467cd2e0c893059b10f78e6b72ada1ed2c919"

    fun saveAccessToken(context: Context, accessToken: String) {
        val cal = Calendar.getInstance()
        cal.add(Calendar.HOUR, 1)
        val oneHourAfter = cal.timeInMillis

        val mSettings = context.getSharedPreferences(BuildConfig.APPLICATION_ID, MODE_PRIVATE)
        val editor = mSettings.edit()

        editor.putString("accessToken", accessToken)
        editor.putLong("expiryDate", oneHourAfter)
        editor.apply()
    }

    fun getAccessToken(context: Context): String? {
        return if (expired(context)) null
        else {
            val mSettings = context.getSharedPreferences(BuildConfig.APPLICATION_ID, MODE_PRIVATE)
            mSettings.getString("accessToken", "")
        }
    }

    private fun expired(context: Context): Boolean {
        val mSettings = context.getSharedPreferences(BuildConfig.APPLICATION_ID, MODE_PRIVATE)
        val expiryTime = mSettings.getLong("expiryDate", 0)
        val currentTime = Calendar.getInstance().timeInMillis
        return currentTime > expiryTime
    }
}